bin/createInhouseSamplesheetFromGS_v2.py

```diff
@@ -0,0 +1,262 @@
+#
+for gsSamplesheetFile in glob.iglob(os.path.join(args.genomeScanInputDir, "UMCG_CSV_*.csv.converted")):
+    if os.path.isfile(gsSamplesheetFile) and os.access(gsSamplesheetFile, os.R_OK):
+        logging.info('Found GenomeScan samplesheet: ' + gsSamplesheetFile + '.')
+    else:
+        logging.critical('Either the ' + args.genomeScanInputDir + 'UMCG_CSV_*.csv.converted file is missing or it is not readable.')
+        sys.exit('FATAL ERROR!')
+
+
+# For debugging data structure only:
+#import pprint
+#pp = pprint.PrettyPrinter(indent=4)
+#pp.pprint(gsFilenameDataHashmap)
+#pp.pprint('=====================================================')
+
+#
+# Combine GS samplesheet with original filename information form gsFilenameDataHashmap.
+#
+# GS samplesheet filestructure for samples prepped not at GenomeScan and send in as pools:
+#          GS_ID,         Sample_ID, Pool,   Index1,   Index2
+# 103473-011-001, QXTR_222-Exoom_v1,    1, CGAGGCTG, AGGCTTAG
+#
+## GS samplesheet filestructure for samples prepped at GenomeScan:
+#          GS_ID,                    ID, positie, geslacht, Pool,   Index1,   Index2
+# 103473-011-001, GS_1A-Exoom_v3-123456,     A01,        V,    1, CGAGGCTG, AGGCTTAG
+#
+gsSamplesheetFileHandle = open(gsSamplesheetFile,'r')
+gsReader = csv.DictReader(gsSamplesheetFileHandle)
+gsHeaders = gsReader.fieldnames
+gsSampleIdColumnName = 'Missing'
+gsSamplesheetDataHashmap = defaultdict(dict)
+gsProjects = []
+if 'Sample_ID' in gsHeaders:
+    gsSampleIdColumnName = 'Sample_ID'
+elif 'ID' in gsHeaders:
+    gsSampleIdColumnName = 'ID'
+else:
+    logging.critical('Cannot find sample ID column name in ' + gsSamplesheetFile + '.')
+    sys.exit('FATAL ERROR!')
+logging.debug('Found sample ID column name ' + gsSampleIdColumnName + ' in ' + gsSamplesheetFile + '.')
+
+for row in gsReader:
+    
+    if row[gsSampleIdColumnName] in (None,""):
+        logging.warning('Empty row detected in GS samplesheet.')
+        continue
+    #
+    # The "sample ID" provided to GenomeScan by our lab is a combination of
+    #  * project
+    #  * sampleProcessStepID; a unique value for each sample.
+    # Use a regular expression to parse the individual values from the combi.
+    #
+    if re.match("(^[a-zA-Z0-9_-]+)-([0-9]+)$", row[gsSampleIdColumnName]):
+        m = re.match("(^[a-zA-Z0-9_-]+)-([0-9]+)$", row[gsSampleIdColumnName])
+        gsProject = m.group(1)
+        gsSampleProcessStepID = m.group(2)
+        logging.debug('Found project ' + gsProject + ' and sampleProcessStepID ' + gsSampleProcessStepID + ' in column ' + gsSampleIdColumnName + '.')
+    else:
+        logging.critical('Cannot parse project name and sampleProcessStepID from "' + row[gsSampleIdColumnName] + '" in column ' + gsSampleIdColumnName + ' from ' + gsSamplesheetFile + '.')
+        sys.exit('FATAL ERROR!')
+    gsGenomeScanID=row['GS_ID']
+    b = re.match("(^[0-9]+-[0-9]+)-([0-9]+)$", row['GS_ID'])
+    gsBatch= b.group(1)
+    gsProjects.append(gsProject)
```

Comment:
No _if-then-else_ block with ```logging.critical(...)``` for this regex means the code will fail in a miserable way without proper error if the regex does not match due to a malformed samplesheet.

Reply:
included an if-then-else block